API/Controllers/UsersController.cs

```diff
@@ -1,37 +1,103 @@
 ﻿using API.Models;
+using API.Models.AccountModels;
 using API.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using System.Threading.Tasks;
 
 namespace API.Controllers
 {
     [Route("api/v1/[controller]/[action]")]
     [Produces(ApiContentType)]
     [ApiController]
+    [Authorize]
     public class UsersController : ApiControllerBase
     {
-        public UsersController(ApiContext context, Iserializer serializer, IErrorFormatter errorFormatter) :
-            base(context, serializer, errorFormatter) { }
+        private readonly UserManager<UserDTO> _userManager;
+        private readonly SignInManager<UserDTO> _signInManager;
+        private readonly JwtService _jwt;
+        public UsersController(
+            ApiContext context,
+            Iserializer serializer,
+            IErrorFormatter errorFormatter,
+            IConfiguration config,
+            UserManager<UserDTO> userManager,
+            SignInManager<UserDTO> signInManager) :
+            base(context, serializer, errorFormatter)
+        {
+            _userManager = userManager;
+            _signInManager = signInManager;
+            _jwt = new JwtService(config);
+        }
 
         [HttpPost]
-        public IActionResult Signup(UserDTO user)
+        [AllowAnonymous]
+        public async Task<IActionResult> Signup(RegisterModel model)
         {
             if (!IsValidApiRequest())
             {
                 return ApiBadRequest("Invalid headers!");
             }
 
-            return NotImplementedError("Signup Not Implemented!");
+            if (ModelState.IsValid)
+            {
+                var user = new UserDTO { UserName = model.Username ?? model.Email, Email = model.Email };
+                var result = await _userManager.CreateAsync(user, model.Password);
+                if (result.Succeeded)
+                {
+                    return Created("", new
```

Comment:
Why is there an empty string? Shouldn't it be something like HttpStatusCode.Created in this case?

Reply:
According to the standard this should be the URI of the resource that has been created, however we don't have that implemented yet